main/src/main/java/cgeo/geocaching/ui/dialog/SimpleDialog.java

```diff
@@ -474,26 +463,83 @@ public final <T> void selectMultiple(@NonNull final List<T> items, @NonNull fina
 
         final AlertDialog dialog = builder.create();
         dialog.show();
-        checkOkButton(dialog, nonEmptyResultRequired, result.size());
+
+        final ArrayAdapter<Integer> adapter = new ArrayAdapter<Integer>(getContext(), 0, 0, filteredPositions) {
+            @NonNull
+            @Override
+            public View getView(final int position, final @Nullable View convertView, final @NonNull ViewGroup parent) {
+                final int originalPos = getItem(position);
+                final View v = convertView != null ? convertView : LayoutInflater.from(getContext()).inflate(R.layout.select_dialog_multichoice_material, parent, false);
+                final CheckedTextView tv = v.findViewById(android.R.id.text1);
+                tv.setText(itemTexts[originalPos]);
+                tv.setChecked(itemSelects[originalPos]);
+
+                tv.setOnClickListener(v1 -> {
+                    final CheckedTextView ctv = (CheckedTextView) v1;
+                    final boolean newStateIsChecked = !ctv.isChecked();
+                    if (addSelectAll && originalPos == 0) {
+                        for (int pos : filteredPositions) {
+                            if (pos != 0) {
+                                if (newStateIsChecked) {
+                                    result.add(items.get(pos - offset));
+                                } else {
+                                    result.remove(items.get(pos - offset));
+                                }
+                            }
+                            itemSelects[pos] = newStateIsChecked;
+                        }
+                        notifyDataSetChanged();
+                    } else {
+                        if (newStateIsChecked) {
+                            result.add(items.get(originalPos - offset));
+                        } else {
+                            result.remove(items.get(originalPos - offset));
+                        }
+                    }
+                    ctv.setChecked(newStateIsChecked);
+                    itemSelects[originalPos] = ctv.isChecked();
+                    checkOkButton(dialog, nonEmptyResultRequired, result.size());
+                });
+                return v;
+            }
+        };
+        lv.setAdapter(adapter);
+
+        ((EditText) dv.findViewById(R.id.filter)).addTextChangedListener(new TextWatcher() {
```

Comment:
Consider using `ViewUtils.createSimpleWatcher()` here

Reply:
Thanks for the pointer. I'll extend this to support `onTextChanged()` as well.

Reply:
I strongl Believe your use case would also work with current implementation (w/o the need to extend to context changed). I use it for example to react on any typing change on formula text field on variables page

Reply:
ah, didn't know that, had only worked with `onTextChanged()` before. I've adapted the PR to work with `afterTextChanged()`.